src/Sylius/Component/Addressing/Matcher/ZoneMatcher.php

```diff
@@ -16,19 +16,41 @@
 use Sylius\Component\Addressing\Model\AddressInterface;
 use Sylius\Component\Addressing\Model\Scope;
 use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Addressing\Model\ZoneMemberInterface;
 use Sylius\Component\Addressing\Repository\ZoneRepositoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
 
 final class ZoneMatcher implements ZoneMatcherInterface
 {
+    private const PRIORITIES = [
+        ZoneInterface::TYPE_PROVINCE,
+        ZoneInterface::TYPE_COUNTRY,
+        ZoneInterface::TYPE_ZONE,
+    ];
+
     /**
-     * @param ZoneRepositoryInterface<ZoneInterface> $zoneRepository
+     * @param RepositoryInterface<ZoneInterface>|ZoneRepositoryInterface<ZoneInterface> $zoneRepository
      */
-    public function __construct(private ZoneRepositoryInterface $zoneRepository)
+    public function __construct(private RepositoryInterface|ZoneRepositoryInterface $zoneRepository)
     {
+        if ($this->zoneRepository instanceof RepositoryInterface) {
```

Comment:
```suggestion
        if (!$this->zoneRepository instanceof ZoneRepositoryInterface) {
```
`ZoneRepositoryInterface` extends `RepositoryInterface`; it'd always trigger